### PR TITLE
fix: zod additional properties error

### PR DESCRIPTION
**Pull request by CxGarcia**
## Status

**READY**

## Description
in zod generator, the `additionalProperties` that are appended to functions array are either an `array` or a `boolean` (see [code](https://github.com/anymaniax/orval/blob/8f1682ce72d721d8992b4c613513191b31029c7c/packages/zod/src/index.ts#L164)). however, `parseProperty` function is treating them all as arrays and using array methods which causes a runtime error. you can reproduce the error by running zod generator on [this file](https://github.com/anymaniax/orval/pull/972/files#diff-2f9624a4143cd200d5a2962c3b6b5e0d0049997d4d2e02ea6d63891ce2716da2)

this PR fixes the error and uses the `passthrough` method when `additionalProperties` are set to `true` to avoid unknown keys being stripped from the object by zod

## Todos

- [x] Tests
- [ ] Documentation (N/A)
- [ ] Changelog Entry (unreleased)

## Steps to Test or Reproduce

```bash
git pull https://github.com/CxGarcia/orval
cd orval
git checkout fix/zod-additional-properties
yarn build
(cd tests && yarn generate:zod)
```

### CODE REVIEW

File: tests/specifications/additional-properties.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 openapi: 3.0.0
 info:
-  title: Circular references
+  title: Addtional properties
```
Comment (melloware) [inline tests/specifications/additional-properties.yaml]:
```suggestion
  title: Additional properties
```

**Review by melloware — CHANGES_REQUESTED** [review]

**Review by melloware — DISMISSED** [review]

**Comment by melloware** [comment]
Looks like tests are failing?

**Review by melloware — CHANGES_REQUESTED** [review]
Tests are still failing...

**Comment by melloware** [comment]
Errors...

```
Error: generated/zod/additionalProperties.ts(18,16): error TS2554: Expected 1-3 arguments, but got 0.
Error: generated/zod/additionalProperties.ts(18,25): error TS2339: Property 'passthrough' does not exist on type 'ZodRecord<ZodString, ZodTypeAny>'.
```

**Comment by soartec-lab** [comment]
Hi, @CxGarcia 
Since our PR was stagnant, created a new PR. You are welcome to participate in this discussion, so please comment here if you have any questions.

https://github.com/anymaniax/orval/pull/1443

If we do not hear from you, we will close this PR and move on to the new one.

**Comment by melloware** [comment]
@soartec-lab i am closing this one.  The new PR is the correct solution.

**Pull request closed without merging**